### ISSUE TITLE
consistent documentation

### ISSUE THREAD

**Issue by iamstarkov** [zeit/hyper#929]
Hyper lacks code and documentation consistency. And It hurts new comers.
- Docs on site are not in sync with actual code: `decorateTerms` and `onRendererLoad` is in the docs, nowhere in the code. Mine and @pakastin's experiences.
- Code not in sync with itself. Hyper API uses `onRendererWindow`, but this method not in the `availableExtensions` list. At the same time, this method is not in the docs. #924

To address it we need:
- docs should state all methods, even deprecated ones. 
- docs should be synced with code, hence jscode or similar tool
- code should aware about itself

TLDR:
- need knowledge about list of existing methods for plugins from Hyper's team.
- make sure docs will not be outdated with the help of jscode or similar tool

**Comment by chabou** [comment]
I was trying to figure out why my `onRendererLoad` function was not called in my plugin...
I search it in Hyper code: not found... 🤔 
But I found this issue 😲 

`decorateTerms` is in code now.
I'll make a PR to replace `onRendererLoad` and `onRendererUnload` by `onRendererWindow`

**Comment by pakastin** [comment]
It's funny (and a bit sad); I got really inspired when [The Hyper Plan](https://zeit.co/blog/the-hyper-plan) came out. Immediately started to fiddle around with Hyper to build my dream dev environment. Not too long after I gave up because nothing worked.

I don't know about the current situation, hope it's better. I'm still using Hyper.app everyday though, just haven't been hacking it at all. Maybe some day I'll try again.

**Comment by chabou** [comment]
I think it is better.
But yes, when you are hacking it (and not just theme-ing), often you discover some issues in Hyper or Docs.

It is because I made plugins that I made some PR like:
#906, #1483, #1494, #1601...

It is a little frustrating but, you know, it is open source and It is up to us to make Hyper even better. Let's hacking it, improve it and have fun :)
